WordPress/src/main/java/org/wordpress/android/ui/sitecreation/NewSiteCreationMainVM.kt

```diff
@@ -41,6 +44,7 @@ class NewSiteCreationMainVM @Inject constructor(
     private val wizardManager: WizardManager<SiteCreationStep>
 ) : ViewModel() {
     private var isStarted = false
+    private var isSitePreviewLayoutShown = false
```

Comment:
I am just curious, are there any alternatives to using this variable to check whether the back button should be suppressed or not? Can we maybe use whether the site creation is completed instead?

I guess this ties into the UX a bit as well, but I feel like if the site is actually created but the preview is not shown it's totally fine for the user to exit the flow without us showing a dialog. Even if we do want to show a dialog, the messaging would probably be different because the user won't use anything by going back.

I think the other major reason I am worried about this approach is that I feel it requires knowledge for why the site preview being shown even matters here. I understand why that's the case since I know the implementation, but just from the main VM, it's not very clear. Maybe the naming change could improve that here, but instead of offering a change here, I thought it'd be more valuable for me to explain what this makes me think so we can come up with something together. (assuming a change is even necessary)

Reply:
>I guess this ties into the UX a bit as well,
This was the main reasoning behind this decision. I felt like we want to show the dialog as long as the fullscreen progress screen is being shown. I've updated the code in [#97b0d1a](https://github.com/wordpress-mobile/WordPress-Android/pull/9187/commits/97b0d1a55fa81eb310569baa46163ae33e0b44c3) and I agree the code is easier to understand + the UX is almost equal.

Reply:
Fixed bug in [#b2baf64](https://github.com/wordpress-mobile/WordPress-Android/pull/9187/commits/b2baf64707ca63f7634e56c5ef4e2c8edd6f88e4)